### DISCUSSION TITLE
How to assert with promises?

### DISCUSSION THREAD

**Discussion by GabrielBarbosaGV** [dubzzz/fast-check#3235] [Q&A] [answered]
Hello, all!

I would like to assert on the promise of a boolean, or, in other words, to assert on an async function returning a boolean. This, for the purpose of testing async code. This is how I am tying it, currently:

![image](https://user-images.githubusercontent.com/29243693/192139002-3d156dd9-365a-4ea5-93a9-f9415533ea3d.png)

This mostly works, but leaves out the important aspect of a failed expectation within the assert callback, which would cause minimization. What happens now is that expect fails, but after assert is already done, which gives me a warning, but no indication of which input caused it. This is an example of that:

![image](https://user-images.githubusercontent.com/29243693/192139206-83e8d251-ff73-4a05-b7da-24819b2ff6db.png)

When naively trying to assert with an async callback, I get a typescript compiler error saying that this is not possible.

How may I assert with promises?

Many thanks to the contributors for this great package!

Kind regards,
Gabriel Barbosa.

**Comment by krisanselmo** [comment, marked as answer]
Hi ! 
If you want to test asynchronous functions, look at `fc.asyncProperty` instead of `fc.property`.
For reference: https://github.com/dubzzz/fast-check/blob/main/packages/fast-check/documentation/Runners.md#properties

**Reply by GabrielBarbosaGV** [reply]
Very quick - many thanks!
